### PR TITLE
generate open fd count without reading fd files

### PR DESCRIPTION
**Pull request by z0br0wn**

### CODE REVIEW

File: process/process_linux.go [hunk 1]
```diff
@@ -220,8 +220,8 @@ func (p *Process) NumCtxSwitches() (*NumCtxSwitchesStat, error) {
 
 // NumFDs returns the number of File Descriptors used by the process.
 func (p *Process) NumFDs() (int32, error) {
-	numFds, _, err := p.fillFromfd()
-	return numFds, err
+	_, fds, err := p.fillFromfdList()
```
Comment (conorbranagan) [inline process/process_linux.go]: can we just push the stuff in `fillFromfdList` into this function since it's not used anywhere else?

**Review by conorbranagan — CHANGES_REQUESTED** [review]
One style comment.

Can we also split this into 2 commits/branches:

1. Where we change NumFDs to just count the len(fds)
2. Where we add this to the FilledProcess

The reason is so you can do a PR upstream for (1) since (IMO) it's valuable to :earth_americas:

File: process/process_linux.go (hunk 1 above)
Comment (z0br0wn) [inline process/process_linux.go]: fillFromfd() still uses the list

Comment (conorbranagan) [inline process/process_linux.go]: ah I can't read, got it. ignore this comment then

**Review by conorbranagan — APPROVED** [review]
:burrito:

**Pull request merged**